src/test/javascript/spec/interceptor/artemis-version.interceptor.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { EMPTY, of, Subject } from 'rxjs';
+import { ServerDateService } from 'app/shared/server-date.service';
+import { ArtemisVersionInterceptor } from 'app/core/interceptor/artemis-version.interceptor';
+import { AlertService } from 'app/core/util/alert.service';
+import { MockService } from 'ng-mocks';
+import { MockArtemisServerDateService } from '../helpers/mocks/service/mock-server-date.service';
+import { discardPeriodicTasks, fakeAsync, tick } from '@angular/core/testing';
+import { HttpHeaders, HttpRequest, HttpResponse } from '@angular/common/http';
+import { SwUpdate } from '@angular/service-worker';
+import { ApplicationRef } from '@angular/core';
+import { ARTEMIS_VERSION_HEADER, VERSION } from 'app/app.constants';
+
+describe(`ArtemisVersionInterceptor`, () => {
+    let appRef: ApplicationRef;
+    let alertService: AlertService;
+    let serverDateService: ServerDateService;
+
+    let swUpdate: any;
+    let checkForUpdateSpy: any;
+    let activateUpdateSpy: any;
+
+    beforeAll(() => {
+        jest.useFakeTimers();
+    });
+
+    afterAll(() => {
+        jest.useRealTimers();
+    });
+
+    beforeEach(() => {
+        swUpdate = {
+            isEnabled: true,
+            activated: EMPTY,
+            available: EMPTY,
+            checkForUpdate: () => of(true).toPromise(),
+            activateUpdate: () => of(true).toPromise(),
```

Comment:
```suggestion
            checkForUpdate: () => Promise.resolve(true),
            activateUpdate: () => Promise.resolve(true),
```
I think this should work to resolve the client style issues.

Reply:
I've adressed this is in 71d127a :)